research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,434 @@
+is "nested" in a currently-shown `popup`; refer to <a href="#hiding-current-popups">hiding
+currently-shown <code>popup</code> elements steps</a>.
+
+<p class="todo">
+  Whenever there is a spec to point to for CSS anchored positioning, call out here that the anchor
+  attribute is used to help resolve CSS <code>anchor()</code> functions.
+</p>
+
+## Hiding a `popup` element
+
+### The `hide()` method
+
+The `hide()` method hides a `popup` element. The `hide()` method steps are to run the
+[hiding a `popup` element steps](#hiding-popup-steps) with
+[this](https://heycam.github.io/webidl/#this).
+
+### Light dismissal
+
+"Light dismiss" describes the behavior of a component which dismisses itself automatically when a
+user performs an action that implies that they are finished interacting with the component. The
+actions that trigger light dismiss are:
+
+- The user presses the escape key.
+- A focus change occurs (because of either user interaction or script), where the focus target is
+  outside of the contents of the component. This includes the case where the user invokes a
+  non-focusable element, which causes focus to jump to the nearest focusable ancestor of that element.
+- An OS-level focus change occurs such that the window containing the component no longer has focus.
+  For example, the user switches to a different browser tab or switches to a different application.
+
+<p class="todo">
+  Land <a href="https://github.com/openui/open-ui/pull/243">#243</a> and link out to this canonical
+  definition of light dismissal for consistency (remove inline definition). Layout changes to the
+  popup or its anchor element are also ground for light dismissal, figure out how to surface this
+  special case.
+</p>
+
+When a `popup` element is shown and a light dismiss interaction occurs:
+
+1. If a keypress event for the `ESCAPE` key was fired, run [hiding a `popup` element
+   steps](#hiding-popup-steps), with the _candidate subject_ set to the top-most `popup` element in
+   the [popup stack](#pop-stack).
+2. If a focus change occurs, the user agent must set the _start node_ to the element which received
+   the focus event and run [Hiding currently-shown `popup` elements steps](#hiding-current-popups).
+3. Otherwise, the user agent must set the _start node_ to null and run [Hiding currently-shown
+   `popup` elements steps](#hiding-current-popups).
+
+<h3 id="hiding-current-popups">
+  Hiding currently-shown <code>popup</code> elements steps
+</h3>
+
+The **hiding currently-shown `popup` elements steps**, given an optional popup element
+_candidate subject_, an optional element _invoker_, and an optional element _anchor element_, and an
+optional element _start node_, are as follows.
+
+For each popup element _subject_ in the
+[node document](https://dom.spec.whatwg.org/#concept-node-document)'s [popup stack](#popup-stack),
+from top to bottom:
+
+1. If _candidate subject_ is null, skip to step 6.
```

Comment:
Like in coding, avoid "goto" in specs. Instead use "If/else". E.g. "If candidate subject is not null: ... nest 5 steps."

It's also not clear what skipping to a step which starts with "Otherwise" means.

Reply:
Addressed in 5fc5041409d28c395e81cf30f039a7c0e269b7fe